delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchIndexingStreamEntry.scala

```diff
@@ -78,14 +79,21 @@ final case class ElasticSearchIndexingStreamEntry(
   def containsTypes[A](resourceTypes: Set[Iri]): Boolean =
     resourceTypes.isEmpty || resourceTypes.intersect(resource.types).nonEmpty
 
-  private def toDocument(includeMetadata: Boolean, sourceAsText: Boolean, context: ContextValue): Task[Json] = {
+  private def toDocument(
+      includeMetadata: Boolean,
+      sourceAsText: Boolean,
+      context: ContextValue,
+      mergeSource: Boolean
```

Comment:
Why is this here? When building the `ElasticSearchIndexingStreamEntry` you can just create the `IndexingData` with an empty source (check the apply methods for it) and then we don't need this, nor yet another if/else

Reply:
I don't understand. You added this for composite views only. sourceAsText does not exist for composite views, so there is nothing to do about that in this case, no?

Reply:
```
/**
    * An ElasticSearch projection for [[CompositeView]].
    */
  final case class ElasticSearchProjection(
      id: Iri,
      uuid: UUID,
      query: SparqlConstructQuery,
      resourceSchemas: Set[Iri],
      resourceTypes: Set[Iri],
      resourceTag: Option[TagLabel],
      includeMetadata: Boolean,
      includeDeprecated: Boolean,
      permission: Permission,
      sourceAsText: Boolean = false,
      mapping: JsonObject,
      settings: Option[JsonObject] = None,
      context: ContextObject
  )
```

Reply:
Thats wrong. Check the v1.4.x docs. And sourceAsText doesn't make any sense for compositeviews

Reply:
https://bluebrainnexus.io/docs/delta/api/current/views/composite-view-api.html#elasticsearchprojection

Reply:
It's not wrong, we decided in the new design to make it consistent with the ElasticsearchView config. Just like the rest of the metadata, while it may not necessarily be useful, we want to offer the option and be consistent.

Reply:
Either way, removing that field is not in the scope of this PR.

Reply:
Ok, then we'll do it right after. 
And yes it is wrong to have sourceAsText in composite views. What you want to index in an elasticsearach view is the source, that's all you have. That's why there is a sourceAsText true|false. 

In composite views you don't have a source, but you have a query result which is transformed to json. There is no room for consistency since an elastic search view and an elastic search projection are not the same thing.